cli/src/commands/branch/set.rs

```diff
@@ -50,7 +50,7 @@ pub fn cmd_branch_set(
     let mut new_branch_names: Vec<&str> = Vec::new();
     for name in branch_names {
         let old_target = repo.view().get_local_branch(name);
-        if old_target.is_absent() {
+        if old_target.is_absent() && !has_tracked_remote_branches(repo.view(), name) {
```

Comment:
Perhaps copy the `// Restoring local target shouldn't invalidate tracking state` comment here?

Reply:
Or maybe it could say something like

> If the branch is missing and tracking remote branches, we are resurrecting the local branch and we shouldn't change it's tracking state.

Reply:
Added comment (but without "we shouldn't change it's tracking state.") Tracking state is unrelated in this context. `new_branch_names` is used only for emitting a warning.